syntax = "proto3";


// Interface exported by the mediator server.

// For the TCP Syslog Source (in the Sources/TCP_Syslog directory):

//  python -m grpc_tools.protoc -I../../proto --python_out=. --grpc_python_out=. ../../proto/mediator.proto

// For the input processor: (in the main source directory):

// python -m grpc_tools.protoc -Iproto --python_out=. --grpc_python_out=. proto/mediator.proto


service Mediator {
  rpc StreamSourceData(stream SourceRecord) returns (Result) {}
  rpc DispatchEvent(EventRecord) returns (Result) {}
  rpc ConnectionTest(Empty) returns (Result) {}
}

message SourceRecord {
  string tenant_id = 1;
  uint64 timestamp = 2;
  int32 type = 3;
  string source_id = 4;
  bytes obj = 5;
}

message EventRecord {
    string tenant_id = 1;
    bytes obj = 2;
}

message Result {
    int32 code = 1;
    string message = 2;
}

message Empty {
}
